app/src/main/java/org/simple/clinic/appupdate/CheckAppUpdateAvailability.kt

```diff
@@ -0,0 +1,65 @@
+package org.simple.clinic.appupdate
+
+import android.app.Application
+import android.os.Build
+import androidx.annotation.VisibleForTesting
+import com.google.android.play.core.appupdate.AppUpdateInfo
+import com.google.android.play.core.appupdate.AppUpdateManagerFactory
+import com.google.android.play.core.install.model.AppUpdateType.FLEXIBLE
+import com.google.android.play.core.install.model.UpdateAvailability
+import io.reactivex.Single
+import org.simple.clinic.util.None
+import org.simple.clinic.util.Optional
+import org.simple.clinic.util.toOptional
+import timber.log.Timber
+import javax.inject.Inject
+
+class CheckAppUpdateAvailability @Inject constructor(
+    private val appContext: Application,
+    private val config: AppUpdateConfig,
```

Comment:
We've always followed the convention of injecting configuration as `Observable<Config>`. Can we do the same here?

Reply:
Since we are not using streams here to get this config, I decided to make as just `AppUpdateConfig`

Reply:
What I mean is that we should be using streams here to get this config since that's what we do everywhere.

Reply:
Will we make this `Observable<AppUpdateConfig>` in a future PR?